RFC-0001-torch-function-for-methods.md

```diff
@@ -0,0 +1,332 @@
+# Improving subclassing Tensor by propagating subclass instances
+This RFC describes changes necessary to allow `__torch_function__` to be used
+by methods of `torch.Tensor` in an attempt to make subclassing more accessible
+to the users of the class. This entails making an API for subclass views
+public, and a change in the signature of `__torch_function__`.
+
+## Motivation and Scope
+Quoting [[1]], [[2]] and [[3]], the goals of this proposal are:
+
+1. Support subclassing `torch.Tensor` in Python
+2. Preserve `Tensor` subclasses when calling `torch` functions on them
+3. Use the PyTorch API with `torch.Tensor`-like objects that are _not_ `Tensor`
+   subclasses
+4. Preserve `Tensor` subclasses when calling `Tensor` methods.
+5. The ability to give external libraries a way to also define
+   functions/methods that follow the `__torch_function__` protocol.
```

Comment:
This one is unrelated, it's kind of a by-product coming from the implementation. It's not a goal, and TBD if actually desired; I'd be inclined to hold off on this.

So that's a separate decision to make. Let's move it out of this list and further down. I'd say expose `torch_function_dispatch` if needed as a tool for subclassing, but let's document for now that it is advised not to be used for anything but subclass methods for now.